modules/library/src/main/scala/zio/elasticsearch/Item.scala

```diff
@@ -16,11 +16,19 @@
 
 package zio.elasticsearch
 
-import zio.json.ast.Json
+import zio.json.ast.{Json, JsonCursor}
 import zio.schema.Schema
 import zio.schema.codec.DecodeError
 import zio.schema.codec.JsonCodec.JsonDecoder
+import zio.{IO, ZIO}
 
-final case class Item(raw: Json) {
+final case class Item(private val raw: Json) {
   def documentAs[A](implicit schema: Schema[A]): Either[DecodeError, A] = JsonDecoder.decode(schema, raw.toString)
+
+  def innerHitsAs[A](name: String)(implicit schema: Schema[A]): IO[DecodingException, List[A]] =
+    for {
+      json     <- ZIO.fromEither(raw.get(JsonCursor.field(name))).mapError(DecodingException)
+      response <- ZIO.fromEither(json.as[SearchWithAggsResponse]).mapError(DecodingException)
+      result   <- new SearchResult(response.results.map(Item)).documentAs[A]
```

Comment:
We would need to chain a few `JsonCursors` to get to the `_source` field, and that is a (mostly) static structure which we mapped to `SearchWithAggsResponse`. For the last line, I just reused `SearchResult` and `Item` to avoid code duplication.